src/CloudNative.CloudEvents.Kafka/KafkaClientExtensions.cs

```diff
@@ -117,42 +145,21 @@ private static void InitPartitioningKey(Message<string, byte[]> message, CloudEv
             var headers = MapHeaders(cloudEvent, formatter);
             string key = (string) cloudEvent[Partitioning.PartitionKeyAttribute];
             byte[] value;
-            string contentTypeHeaderValue = null;
+            string contentTypeHeaderValue;
 
-            if (contentMode == ContentMode.Structured)
+            switch (contentMode)
             {
-                value = formatter.EncodeStructuredModeMessage(cloudEvent, out var contentType);
-                // TODO: What about the non-media type parts?
-                contentTypeHeaderValue = contentType.MediaType;
-            }
-            else
-            {
-                if (cloudEvent.Data is byte[] byteData)
-                {
-                    value = byteData;
-                }
-                else if (cloudEvent.Data is Stream dataStream)
-                {
-                    // TODO: Extract this common code somewhere, or use shared source to access BinaryDataUtilities.
-                    if (dataStream is MemoryStream dataMemoryStream)
-                    {
-                        value = dataMemoryStream.ToArray();
-                    }
-                    else
-                    {
-                        var buffer = new MemoryStream();
-                        dataStream.CopyTo(buffer);
-                        value = buffer.ToArray();
-                    }
-                }
-                else
-                {
-                    throw new InvalidOperationException($"{cloudEvent.Data.GetType()} type is not supported for Cloud Event's Value.");
-                }
-                if (cloudEvent.DataContentType is string dataContentType)
-                {
-                    contentTypeHeaderValue = dataContentType;                    
-                }
+                case ContentMode.Structured:
+                    value = formatter.EncodeStructuredModeMessage(cloudEvent, out var contentType);
+                    // TODO: What about the non-media type parts?
```

Comment:
would something like the charset get included in contentType.ToString()? Would it make sense to use that instead of MediaType?

Reply:
Yes, I suspect that's the way forward. I've just been reluctant to do it so far, just in case it was actually deliberate. (It's in a bunch of places - I think we'll probably want to fix them all in one pass.)